src/schemathesis/models.py

```diff
@@ -178,8 +179,12 @@ def call(
             close_session = False
 
         data = self.as_requests_kwargs(base_url)
+        if "headers" not in data or data["headers"] is None:
+            data.update({"headers": {}})
         if headers is not None:
-            data["headers"] = {**(data["headers"] or {}), **headers}
+            data["headers"].update(headers)
+        if "User-Agent" not in data["headers"]:
```

Comment:
Our "headers" value doesn't count with case sensitivity and if that dictionary contains "user-agent", then I am not sure which variant will be used by requests, but apparently it will be the last value:

```
In [2]: requests.get("https://httpbin.org/get", headers={"user-agent": "foo", "User-Agent": "bar"}).json()                                                                                                                                                                                                                                              
Out[2]: 
{'args': {},
 'headers': {'Accept': '*/*',
  'Accept-Encoding': 'gzip, deflate',
  'Host': 'httpbin.org',
  'User-Agent': 'bar',
  'X-Amzn-Trace-Id': 'Root=1-5f7585c3-5603d8fe64303f8e0441b5c7'},
 'origin': '0.0.0.0',
 'url': 'https://httpbin.org/get'}
```

We can leave it as it is for now and solve case-sensitivity in a separate issue since it also involves code in the "loaders" module and the "runner" package. In general I consider it as a bug and will create a bug report for it.

Reply:
I'd rather limit the scope of this PR and solve #757 separately